server/server.go

```diff
@@ -78,7 +78,13 @@ func (h *serverAuthProxyHandler) handler(w http.ResponseWriter, r *http.Request)
 	statusCode := http.StatusOK
 
 	// before we handle any requests, make sure the user is authenticated
-	if h.credentials.Username != "" || h.credentials.Password != "" {
+	if r.Header.Get("Kiali-Token") != "" {
```

Comment:
JWT is usually used (not a requirement, but i have seen that most services do this way) in the header 'Authorization' and usually has the format "Bearer the_token"
See [jwt.io](https://jwt.io/introduction/) [1] and [jira](https://developer.atlassian.com/cloud/jira/software/oauth-2-jwt-bearer-token-authorization-grant-type/#using-an-access-token-in-a-request) docs, [wikipedia](https://en.wikipedia.org/wiki/JSON_Web_Token#Use), etc.

I think we should comply unless there is a reason we can't do that.

[1]
> Whenever the user wants to access a protected route or resource, the user agent should send the JWT, typically in the Authorization header using the Bearer schema. The content of the header should look like the following:
> 
> Authorization: Bearer \<token\>